cmake/cxxflags_common.cmake

```diff
@@ -0,0 +1,80 @@
+
+
+# This is unfortunately still needed to disable exceptions/RTTI since modern CMake still has no builtin support...
+# E.g. replace_cxx_flag("/EHsc", "/EHs-c-")
+macro(replace_cxx_flag pattern text)
+    foreach (flag
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+
+        string(REGEX REPLACE "${pattern}" "${text}" ${flag} "${${flag}}")
+
+    endforeach()
+endmacro()
+
+# Fixup default compiler settings
+if (MSVC)
+    add_compile_options(
+        # Be as strict as reasonably possible, since we want to support consumers using strict warning levels
+        /W4 /WX
+        )
+else()
+    # Clang with non-MSVC commandline syntax
+    add_compile_options(
+        # Effectively the same as /W4 /WX
+        -Wall -Werror
+    )
+endif()
+
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
Is this how you're supposed to tell if you're using CLANG?
You'd probably need some additional flag to tell whether you are using CLANG on Mac or on Linux.